src/GitHub.Exports/Models/DiffUtilities.cs

```diff
@@ -113,6 +114,65 @@ public static DiffLine Match(IEnumerable<DiffChunk> diff, IList<DiffLine> target
             return null;
         }
 
+        public class LineReader
+        {
+            readonly string text;
+            int index = 0;
+
+            public LineReader(string text)
+            {
+                this.text = text;
+            }
+
+            public string ReadLine()
+            {
+                if (EndOfText)
+                {
+                    if (StartOfText)
+                    {
+                        index = -1;
+                        return string.Empty;
+                    }
+
+                    return null;
+                }
+
+                var startIndex = index;
+                index = text.IndexOf('\n', index);
+                var endIndex = index != -1 ? index : text.Length;
+                var length = endIndex - startIndex;
+
+                if (index != -1)
+                {
+                    if (index > 0 && text[index - 1] == '\r')
+                    {
+                        length--;
+                    }
+
+                    index++;
+                }
+
+                return text.Substring(startIndex, length);
+            }
+
+            bool StartOfText => index == 0;
+
+            bool EndOfText => index == -1 || index == text.Length;
```

Comment:
If null is passed in to the LineReader construction, this will throw.

Reply:
Added null check to constructor.